api.go

```diff
@@ -519,15 +519,21 @@ func (c Client) executeMethod(method string, metadata requestMetadata) (res *htt
 	// Indicate to our routine to exit cleanly upon return.
 	defer close(doneCh)
 
+	// Retry only when reader is seekable
+	reqRetry := MaxRetry
+	if !isSeekable {
+		reqRetry = 1
```

Comment:
We should set this value perhaps in above `metadata.contentBody` fold itself.  Avoids having multiple ifs.